Python/ceval.c

```diff
@@ -5069,6 +5070,86 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             DISPATCH();
         }
 
+        TARGET(CALL_NO_KW_SUPER_0__LOAD_METHOD_CACHED) {
+            /* super().meth */
+            assert(_Py_OPCODE(next_instr[0]) == LOAD_METHOD_ADAPTIVE);
+            assert(_Py_OPCODE(next_instr[-2]) != PRECALL_METHOD);
+            SpecializedCacheEntry *caches = GET_CACHE();
+            _PyAdaptiveEntry *cache0 = &caches[0].adaptive;
+            _PyObjectCache *cache1 = &caches[-1].obj;
+            _PyAdaptiveEntry *lm_adaptive = &caches[-2].adaptive;
```

Comment:
Is this the cache entry for the following `LOAD_METHOD`?
If it is, can you add an assert that this is the correct cache entry for the `LOAD_METHOD`?

Reply:
Yes it is. I'll add an assert.

I'm aware it's a little risky to use the next cache entry. If the hypothetical situation where CALL_SUPER_LOAD_METHOD deopts, and the next LOAD_METHOD then specializes, the subsequent call to CALL_SUPER_LOAD_METHOD will at best de-opt again, at worst erroneously allow invalid conditions to execute.

To protect against this, we need to ensure LOAD_METHOD will never specialize for `super`, that way its memory will remain unused (and usable by CALL_SUPER_LOAD_METHOD.